### PR TITLE
[Lint] Update linter to use babel-lint and lint JSX/React directly

### PR DESCRIPTION
**Pull request by ide**
- Upgraded eslint
- Installed babel-eslint so that we can parse JSX and ES6
- Installed eslint-react-plugin so we can lint JSX directly w/o a transform
- `no-comma-dangle` was replaced with `comma-dangle`
- `space-unary-word-ops` was replaced with `space-unary-ops`.

### CODE REVIEW

**Comment by sahrens** [comment]
@amasad Can we do this yet?

**Comment by amasad** [comment]
I think we need to update our eslint version first. Right?

**Comment by ide** [comment]
Probably the best way to approach this is to install eslint, babel-lint, and eslint-react-plugin directly so that React Native can have its own linter settings independent of any other project. My latest commit does this but there's an issue with the packager where it's finding conflicting `@providesModule` directives deep inside of node_modules (the "commoner" module's files) so that needs to be resolved before merging.

**Comment by amasad** [comment]
I had this on my list of things todo. I'll take a look soon!

**Comment by amasad** [comment]
Didn't forget about this. Switching to babel very soon and then we can easily use this.

**Comment by ide** [comment]
I'll keep rebasing periodically so it's easy to merge and uses the latest eslint plugins.

**Comment by amasad** [comment]
Thanks @ide! We just switched to Babel, and we can safely merge this now (after testing internally, with phabricator etc)
@jaredly will take care of this.

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "debug": "~2.1.0",
     "graceful-fs": "^3.0.6",
     "image-size": "0.3.5",
+    "jest-cli": "^0.4.3",
```
Comment (jaredly) [inline package.json]: Was this intentional?

Comment (ide) [inline package.json]: it's unrelated iirc

Comment (amasad) [inline package.json]: jest should only be in devDependencies

**Comment by ide** [comment]
Removed the unnecessary jest-cli entry from dependencies.

**Comment by jaredly** [comment]
Thanks!
I had to make some changes to get this to work for both open source and internal, and will show up at the next sync.

**Comment by ide** [comment]
Sweet, looking forward to it.

**Comment by ide** [comment]
Yay! Thanks for landing it =)

**Comment by amasad** [comment]
:dancer:

**Pull request closed without merging**